docs/docs/gatsby-lifecycle-apis.md

```diff
@@ -13,4 +13,49 @@ Gatsby's design principles include:
   trivial and encouraged.
 - Plugins are easy to open source and reuse. They're just NPM packages.
 
-See the links along the left under "REFERENCE" for the full API documentation.
+# High level Overview
+
+## Bootstrap sequence
+
+During "bootstrap" gatsby:
+
+- reads `gatsby-config.js` to load in your list of plugins
+- initializes its cache (stored in `/.cache`)
+- pulls in and preprocesses data ("source and transform nodes") into a GraphQL schema
+- creates pages in memory
+  - from your `/pages` folder 
+  - from your `gatsby-node.js` if you implement `createPages`/`createPagesStatefully` (e.g. templates)
+  - from any plugins that implement `createPages`/`createPagesStatefully`
+- extracts, runs, and replaces graphql queries for pages and `StaticQuery`s
+- writes out the pages to cache
+
+In development this is a running process powered by [Webpack](https://github.com/gatsbyjs/gatsby/blob/dd91b8dceb3b8a20820b15acae36529799217ae4/packages/gatsby/package.json#L128) and [react-hot-loader](https://github.com/gatsbyjs/gatsby/blob/dd91b8dceb3b8a20820b15acae36529799217ae4/packages/gatsby/package.json#L104), so changes to any files get re-run through the sequence again, with [smart cache invalidation](https://github.com/gatsbyjs/gatsby/blob/ffd8b2d691c995c760fe380769852bcdb26a2278/packages/gatsby/src/bootstrap/index.js#L141). In particular, source plugins like gatsby-source-filesystem can also be watching files for changes which can trigger re-running queries. Queries are also watched so if you modify a query your development environment is reloaded.
+
```

Comment:
> The can words make it sound like source re-running queries only gets triggered sometimes.

ah well I mean it really depends on what plugins you have. I think what the phrase is trying to express is that if you have a plugin like `gatsby-source-filesystem` it -will- watch the files for changes, but if you have other plugins it might not. so its deterministic based on the plugin, but its not like we're not gonna know what plugins you have. how best to express this? I don't know. personally from working with Gatsby enough I don't even feel like this is an important detail, I just always assume the queries re-run.

Reply:
>  always uses backticks 

ok sure. I will address this once we finalize the other comment, didn't want GitHub to swallow it while its still unaddressed